### PR TITLE
Calculator: Does not handle log bases correctly

### LINKED ISSUE #4158: Calculator: Does not handle log bases correctly
log2(2) returns 0
Should be returning 1

------
IA Page:  http://duck.co/ia/view/calculator 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @pjhampton

### PR DESCRIPTION
**Pull request by lernae**
## Related Issues and Discussions

Fixes #4158 Calculator: Does not handle log bases correctly
This change updates perl triggering on "log" to only match log base 10.  
Related test cases are included to cover log10 and non log10 input.

## People to notify

@pjhampton 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Calculator.pm [hunk 1]
```diff
@@ -18,7 +18,7 @@ triggers query_nowhitespace => qr'^
     \d+\%=?$ |
     what is| calculat(e|or) | solve | math |
     times | divided by | plus | minus | cos | tau |
-    sin | tan | cotan | log | ln | exp | tanh |
+    sin | tan | cotan | log10\b | ln | exp | tanh |
```
Comment (mintsoft) [inline lib/DDG/Goodie/Calculator.pm]: We probably want `log | log10\b | ln` so that `log` works?

**Comment by lernae** [comment]
hi @mintsoft - Thanks for reviewing this.  My understanding is that this is only a temporary change as currently there is a bug in JS expression normalizer where only log base 10 returns correct results.  Accordingly, this fix is updating the perl trigger to only allow the log base 10 input.  @pjhampton is planning to look into fixing  the expression normalizer for log input in JS in the upcoming weeks, at which point, I expect this change to be reverted.

**Comment by pjhampton** [comment]
Hey @lernae 
Sorry about this, but we're just going to rewrite this logic now. I really do appreciate you weighing in though ❤️  If you want some more high impact stuff to work on then please reach out 😄 

p

**Pull request closed without merging**